components/sup/src/main.rs

```diff
@@ -73,6 +73,8 @@ static RING_ENVVAR: &'static str = "HAB_RING";
 static RING_KEY_ENVVAR: &'static str = "HAB_RING_KEY";
 
 fn main() {
+    env_logger::init();
+    enable_features_from_env();
```

Comment:
I can't see any reason for these to not happen right away. Doing it later removes the ability to do tunable logging or depend upon features.